utils/filter.c

```diff
@@ -1687,4 +1687,542 @@ TEST_CASE(trigger_setup_args)
+	TEST_EQ(filter->start, 0x1000UL);
+	TEST_EQ(filter->end, 0x1000UL + 0x1000UL);
+	uftrace_cleanup_filter(&root);
+	TEST_EQ(RB_EMPTY_ROOT(&root), true);
+
+	pr_dbg("checking unknown symbol\n");
+	uftrace_setup_loc_filter("invalid_name", &sinfo, &root, NULL, &setting);
+	TEST_EQ(RB_EMPTY_ROOT(&root), true);
+
+	return TEST_OK;
+}
+
+TEST_CASE(locfilter_setup_regex)
+{
+	struct uftrace_sym_info sinfo = {
+		.loaded = false,
+	};
+	struct rb_root root = RB_ROOT;
+	struct rb_node *node;
+	struct uftrace_filter *filter;
+	struct uftrace_filter_setting setting = {
+		.ptype = PATT_REGEX,
+	};
+
+	locfilter_test_load_symtabs(&sinfo);
+
+	pr_dbg("try to match with regex pattern: re.*\n");
+	uftrace_setup_loc_filter("re.*", &sinfo, &root, NULL, &setting);
+	TEST_EQ(RB_EMPTY_ROOT(&root), false);
+
+	node = rb_first(&root);
+	filter = rb_entry(node, struct uftrace_filter, node);
+	TEST_STREQ(filter->name, "command_replay");
+	TEST_EQ(filter->start, 0x2000UL);
+	TEST_EQ(filter->end, 0x2000UL + 0x1000UL);
+
+	node = rb_next(node);
+	filter = rb_entry(node, struct uftrace_filter, node);
+	TEST_STREQ(filter->name, "command_report");
+	TEST_EQ(filter->start, 0x3000UL);
+	TEST_EQ(filter->end, 0x3000UL + 0x1000UL);
+
+	node = rb_next(node);
+	filter = rb_entry(node, struct uftrace_filter, node);
+	TEST_STREQ(filter->name, "util_report");
+	TEST_EQ(filter->start, 0xb000UL);
+	TEST_EQ(filter->end, 0xb000UL + 0x1000UL);
+
+	TEST_EQ(rb_next(node), NULL);
+
+	pr_dbg("found 3 symbols. done\n");
+	uftrace_cleanup_filter(&root);
+	TEST_EQ(RB_EMPTY_ROOT(&root), true);
+
+	return TEST_OK;
+}
+
+TEST_CASE(locfilter_setup_glob)
+{
+	struct uftrace_sym_info sinfo = {
+		.loaded = false,
+	};
+	struct rb_root root = RB_ROOT;
+	struct rb_node *node;
+	struct uftrace_filter *filter;
+	struct uftrace_filter_setting setting = {
+		.ptype = PATT_GLOB,
+	};
+
+	locfilter_test_load_symtabs(&sinfo);
+
+	pr_dbg("try to match with glob pattern: re.*\n");
+	uftrace_setup_loc_filter("*re*", &sinfo, &root, NULL, &setting);
+	TEST_EQ(RB_EMPTY_ROOT(&root), false);
+
+	node = rb_first(&root);
+	filter = rb_entry(node, struct uftrace_filter, node);
+	TEST_STREQ(filter->name, "command_replay");
+	TEST_EQ(filter->start, 0x2000UL);
+	TEST_EQ(filter->end, 0x2000UL + 0x1000UL);
+
+	node = rb_next(node);
+	filter = rb_entry(node, struct uftrace_filter, node);
+	TEST_STREQ(filter->name, "command_report");
+	TEST_EQ(filter->start, 0x3000UL);
+	TEST_EQ(filter->end, 0x3000UL + 0x1000UL);
+
+	node = rb_next(node);
+	filter = rb_entry(node, struct uftrace_filter, node);
+	TEST_STREQ(filter->name, "util_report");
+	TEST_EQ(filter->start, 0xb000UL);
+	TEST_EQ(filter->end, 0xb000UL + 0x1000UL);
+
+	TEST_EQ(rb_next(node), NULL);
+
+	pr_dbg("found 3 symbols. done\n");
+	uftrace_cleanup_filter(&root);
+	TEST_EQ(RB_EMPTY_ROOT(&root), true);
+
+	return TEST_OK;
+}
+
+/* Simple pattern arguments are treated as regex after conversion. */
+TEST_CASE(locfilter_setup_directory_simple)
```

Comment:
I'll merge this with locfilter_setup_directory_regex if you want.
Thank you.

Reply:
I think it's good to leave as is.  But I prefer short names like 'locfilter_setup_dir_simple'.